modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -620,8 +619,12 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         }
 
         if (currentTime > experiment.getEndTime().getTime()) {
+            // the experiment moves to the PAUSED state if the endtime is reached
+            experiment.setState(Experiment.State.PAUSED);
+            experimentUtil.updateExperimentState(experiment, Experiment.State.PAUSED);
+            metadataCache.refresh();
             return nullAssignment(userID, applicationName, experimentID,
```

Comment:
@AndreaSuckro Existing behavior: For PAUSED experiment, system returns EXISTING_ASSIGNMENT assignment if user was assigned to the experiment or returns EXPERIMENT_PAUSED assignment.

Can you think of maintaining the same behavior? 

If NULL-Assignment (EXPERIMENT_PAUSED) is returned from here then above behavior will break...